relayer/processor/codec.go

```diff
@@ -0,0 +1,47 @@
+package processor
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	sdkcodec "github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
+	ibc "github.com/cosmos/ibc-go/v7/modules/core"
+	solotypes "github.com/cosmos/ibc-go/v7/modules/light-clients/06-solomachine"
+	tmtypes "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+)
+
+var modBasics = []module.AppModuleBasic{
+	ibc.AppModuleBasic{},
+	tmtypes.AppModuleBasic{},
+	solotypes.AppModuleBasic{},
+}
+
+type codec struct {
+	InterfaceRegistry types.InterfaceRegistry
+	Marshaler         sdkcodec.Codec
+	TxConfig          client.TxConfig
+	Amino             *sdkcodec.LegacyAmino
+}
+
+func makeCodec() codec {
```

Comment:
I think we can simplify the entire codec to:

```
import (
	sdkcodec "github.com/cosmos/cosmos-sdk/codec"
	"github.com/cosmos/cosmos-sdk/codec/types"
	tmtypes "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
)

func tmClientCodec() *sdkcodec.ProtoCodec {
	interfaceRegistry := types.NewInterfaceRegistry()
	tmtypes.RegisterInterfaces(interfaceRegistry)
	return sdkcodec.NewProtoCodec(interfaceRegistry)
}
```

And then store this as a global codec `var TMClientCodec = tmClientCodec()` in either the provider or pathEndRuntime, then use `TMClientCodec`  in `UnmarshalClientMessage`